webapp/channels/src/components/post_view/post_reaction/post_reaction.tsx

```diff
@@ -53,6 +53,12 @@ export default class PostReaction extends React.PureComponent<Props, State> {
         this.props.toggleEmojiPicker();
     };
 
+    toggleEmojiPicker = (e?: React.ChangeEvent<HTMLInputElement>): void => {
+        if(!e?.target.getElementsByClassName('close emoji-picker__header-close-button').length) {
+            this.props.toggleEmojiPicker();
+        }
```

Comment:
can you add a comment here on why this `getElementsByClassName` exists here?

Reply:
I'd like to know that as well. Generally, `getElementsByClassName` is something we should only be using when we need to interact with the DOM directly, but in this case, it looks like we might be using it to check for some view state that we get another way

Reply:
Done